### PR TITLE
Feature/channel summation

### PR DESCRIPTION
**Pull request by psurukuc**
Not fully ready to merge. Need to confirm comments and such.

### CODE REVIEW

**Comment by psurukuc** [comment]
Andrew has been implementing his time-dependent grid in Katydid and that depends on some of the updates on this branch. Plan to merge this branch to develop and then push those changes into feature/time-domain-ch-sum branch before requesting a PR.

File: Source/SpectrumAnalysis/KTChannelAggregator.cc
Comment (nsoblath) [inline Source/SpectrumAnalysis/KTChannelAggregator.cc:26]: Values should be initialized to things that are valid values for those variables, but will generally _not_ be useful in an analysis.  This way all users are made to explicitly specify things in their configs and not rely on default values that might get changed. So, e.g. numbers of things initialized to 0; maxes and mins to 0 and 1, respectively; wavelength to 0.

File: Source/SpectrumAnalysis/KTChannelAggregator.cc
Comment (nsoblath) [inline Source/SpectrumAnalysis/KTChannelAggregator.cc:104]: `int` --> `unsigned` and spaces between arguments

File: Source/SpectrumAnalysis/KTChannelAggregator.cc
Comment (nsoblath) [inline Source/SpectrumAnalysis/KTChannelAggregator.cc:119]: `int` --> `unsigned` and spaces between arguments

File: Source/SpectrumAnalysis/KTChannelAggregator.cc
Comment (nsoblath) [inline Source/SpectrumAnalysis/KTChannelAggregator.cc:207]: `int` --> `unsigned` and more spaces.
Either `if (true)` or `if( true )`, but be consistent.

File: Source/SpectrumAnalysis/KTChannelAggregator.hh
Comment (nsoblath) [inline Source/SpectrumAnalysis/KTChannelAggregator.hh:89]: unsigned?

File: Source/SpectrumAnalysis/KTChannelAggregator.hh
Comment (nsoblath) [inline Source/SpectrumAnalysis/KTChannelAggregator.hh:101]: unsigned

File: Source/SpectrumAnalysis/KTChannelAggregator.hh
Comment (nsoblath) [inline Source/SpectrumAnalysis/KTChannelAggregator.hh:120]: return value `unsigned`?

File: Source/SpectrumAnalysis/KTChannelAggregator.hh
Comment (nsoblath) [inline Source/SpectrumAnalysis/KTChannelAggregator.hh:12]: This file has a bunch of tab characters in it.  Please convert all to 4 spaces

**Review by nsoblath — CHANGES_REQUESTED** [review]
Very nice overall.  I didn't check all of the math, or anything, but for the most part the changes looked consistent with best practices and standards, both for C++ in general and Katydid specifically.  

Is there a test program for the channel aggregators?  If not, please add one that verifies that the processors are behaving as expected.

I have some comments that apply in many places, and I've tried to include individual comments where I've noticed individual instances, but some may have been missed.

* There are a number of places where spaces are missing between function arguments (calls and definitions) or where multiple variables are defined in one line.  Please add those spaces.
* Make sure to initialize variables where they're defined to avoid the possibility that they're used uninitialized.
* Standard practice in Katydid is to use unsigned ints for counting variables (like components or bins) if there's no chance of using a negative number.  In some cases I was pretty sure unsigned variables should be used.  In some cases I wasn't sure whether or not there was a possibility of a negative value.  Please check these and use unsigned in any case where negative values aren't expected.
* There were a few hard tab characters in at least one file.

File: Examples/ConfigFiles/GridPoints.txt
Comment (psurukuc) [reply Examples/ConfigFiles/GridPoints.txt:1]: Hi Noah, yes, there is no compelling reason to use text file. I work on chaging this to YAML file. Just so I unserstand correctly, you want that to be done in a new feature branch right after merging this one into develop ?

Comment (nsoblath) [reply Examples/ConfigFiles/GridPoints.txt:1]: That's right.  As I mentioned in one of the other comments, let's rename the relevant config parameters so they don't say "text" in their names, in anticipation of the change.  But let's not hold up the merge for this particular change.

File: Source/IO/Conversions/KT2ROOT.cc
Comment (psurukuc) [reply Source/IO/Conversions/KT2ROOT.cc:490]: Done

File: Source/IO/Conversions/KT2ROOT.cc
Comment (psurukuc) [reply Source/IO/Conversions/KT2ROOT.cc:501]: Done

File: Source/IO/BasicROOTFileWriter/KTBasicROOTTypeWriterSpectrumAnalysis.cc
Comment (psurukuc) [reply Source/IO/BasicROOTFileWriter/KTBasicROOTTypeWriterSpectrumAnalysis.cc:880]: Done

File: Source/SpectrumAnalysis/KTAxialAggregator.cc
Comment (psurukuc) [reply Source/SpectrumAnalysis/KTAxialAggregator.cc:35]: That is a mistake, thanks for catching.

**Comment by psurukuc** [comment]
Hi Noah, thanks for very carefully going through the commits. I think I fixed all the suggestions expect for replacing YAML file with text file.

**Comment by psurukuc** [comment]
Actually, please hold off on reviewing. I think I may have to make a few more changes. Also, I realized I haven't actually pushed all my changes to github yet.

**Comment by psurukuc** [comment]
@nsoblath, thanks for the comments. I have implemented all the changes. There is a test program called TestChannelAggregator to test the ChannelAggregator and I modified it slightly to fit the new changes better.

**Review by nsoblath — APPROVED** [review]
This all looks good.  Thanks for all of the efforts on the this PR!

**Pull request merged**
